internal/decryption/decryption.go

```diff
@@ -34,19 +33,19 @@ type KeyProviderOpts struct {
 // Decrypt decrypts the content of an audit log file using the metadata found in the file,
 // the credentials provided by the user and the AES-GCM algorithm.
 // The decrypted audit log records are saved in the out stream.
-func Decrypt(logReader io.Reader, out io.Writer, opts KeyProviderOpts) error {
-	auditLogFormat, logLines, err := readAuditLogFile(logReader)
+func Decrypt(logReader io.ReadSeeker, out io.Writer, opts KeyProviderOpts) error {
+	_, logLineScanner, err := readAuditLogFile(logReader)
 	if err != nil {
 		return err
 	}
 
-	auditLogEncoding := newAuditLogEncoding(auditLogFormat)
 	output := buildOutput(out)
 	var decryptSection *DecryptSection
 
-	for idx, line := range logLines {
+	idx := 0
+	for ; logLineScanner.Scan(); idx++ {
 		lineNb := idx + 1
-		logLine, err := auditLogEncoding.Parse(line)
+		logLine, err := logLineScanner.AuditLogLine()
```

Comment:
@tibulca @blva I had to move things around because I was not counting for the case when one line fails to parse and we continue to the next line